joblib/parallel.py

```diff
@@ -179,15 +188,33 @@ def __call__(self):
     def __len__(self):
         return self._size
 
+    @staticmethod
+    def _wrap_non_picklable_objects(obj):
```

Comment:
Please insert a comment explaining why we do this shallow introspection of objects instead of using the `cloudpickle.Pickler` class directly to pickle everything.

```python
# The following does a shallow introspection of the Python objects passed as
# args to a parallel call. Interactively defined functions are pre-pickled manually
# using cloudpickle.
# We do not use cloudpickle for all arguments as it would incur a significant
# performance overhead when pickling large cardinality dicts or lists for
# instance.
```